app/models/duration_rate.rb

```diff
@@ -2,10 +2,26 @@
 
 class DurationRate < ApplicationRecord
 
-  belongs_to :instrument, foreign_key: :product_id
+  belongs_to :price_group
+  belongs_to :rate_start, required: true
 
-  validates :min_duration, presence: true
-  validates :min_duration, numericality: { greater_than_or_equal_to: 0, allow_blank: true }
-  validates :rate, presence: true
+  validate :rate_or_subsidy
   validates :rate, numericality: { greater_than_or_equal_to: 0, allow_blank: true }
+  validates :subsidy, numericality: { greater_than: 0, allow_blank: true }
+
+  # TODO: Validate:
+  # - Rate is <= base rate (if Price group is the “base” group or external).
+  # - Subsidy is <= base rate (if Price group is internal but not the “base”).
+  # For this it looks like we'll need to link Rate Start to Price Policy, instead of Instrument.
+  # In this context we only have a Price Group (which can have N Price policies) and an Instrument (which can have N Price policies).
+  # We need a Price Policy as that's where we store the usage_rate.
+  # Also, makes sense that Steps for an Instrument can change in time but we may want to keep the historic data.
```

Comment:
We're going to log changes to the Event Log, any other historic tracking is out of scope for now.